dds/DCPS/DiscoveryBase.h

```diff
@@ -1009,6 +1009,10 @@ namespace OpenDDS {
           discovered_endpoints = td.discovered_subscriptions();
         }
 
+        if (local_endpoints.empty()) {
+          return;
+        }
```

Comment:
It seems that when this function is called `local_endpoints` can be empty and `repoId` can be a local endpoint that has to make it to the `discovered_endpoints` for loop or else it causes problems.